art/preprocessing/standardisation_mean_std/standardisation_mean_std_pytorch.py

```diff
@@ -69,14 +73,16 @@ def forward(
     ) -> Tuple["torch.Tensor", Optional["torch.Tensor"]]:
         """
         Apply standardisation with mean and standard deviation to input `x`.
+
+        :param x: Input data for which the gradient is estimated. First dimension is the batch size.
+        :param grad: Gradient value so far.
+        :return: The gradient (estimate) of the defence.
         """
         import torch  # lgtm [py/repeated-import]
 
-        mean = torch.tensor(self.mean, device=self._device)
-        std = torch.tensor(self.std, device=self._device)
-
-        x_norm = x - mean
-        x_norm = x_norm / std
+        mean, std = broadcastable_mean_std(x, self.mean, self.std)
```

Comment:
Would it be possible to run broadcastable_mean_std outside of method forward? It should only be required once.